### PR TITLE
NumPy array creation routines

### PR DESCRIPTION
**Pull request by alexnick83**
Adds support for the following NumPy array creation routines:
- empty
- empty_like
- ones
- ones_like
- zeros
- zeros_like
- full
- full_like
- copy
- identity

### CODE REVIEW

File: dace/frontend/python/replacements.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
+from sympy.simplify.fu import L
```
Comment (tbennun) [inline dace/frontend/python/replacements.py:2]: pylance?

File: dace/frontend/python/replacements.py [hunk 2]
```diff
@@ -210,6 +211,165 @@ def eye(pv: 'ProgramVisitor',
...
+def _numpy_copy(pv: 'ProgramVisitor',
+                sdfg: SDFG,
+                state: SDFGState,
+                a: str):
+    """ Creates a copy of array a.
+    """
+    if a not in sdfg.arrays.keys():
+        raise mem_parser.DaceSyntaxError(
+            pv, None, "Prototype argument {a} is not SDFG data!".format(a=a))
+    # TODO: The whole AddTransientMethod class should be move in replacements.py
```
Comment (tbennun) [inline dace/frontend/python/replacements.py:364]: Don't we have `sdfg.add_temp_transient` for that?

**Review by tbennun — CHANGES_REQUESTED** [review]
Nice! Two very minor changes

File: dace/frontend/python/replacements.py (hunk 1 above)
Comment (alexnick83) [reply dace/frontend/python/replacements.py:2]: Ohdeeralarm

File: dace/frontend/python/replacements.py (hunk 2 above)
Comment (alexnick83) [reply dace/frontend/python/replacements.py:364]: This methods handles different types of data (scalar, array, stream) The SDFG class method is only for arrays, right?

File: dace/frontend/python/replacements.py (hunk 1 above)
Comment (tbennun) [inline dace/frontend/python/replacements.py:2]:
```suggestion
```

File: dace/frontend/python/replacements.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
 from dace.frontend.common import op_repository as oprepo
 import dace.frontend.python.memlet_parser as mem_parser
 from dace.frontend.python.memlet_parser import parse_memlet_subset
-from dace.frontend.python import astutils
+from dace.frontend.python import astutils, newast
```
Comment (tbennun) [inline dace/frontend/python/replacements.py:19]:
```suggestion
from dace.frontend.python import astutils
```

**Review by tbennun — APPROVED** [review]

**Pull request merged**